plugins/catalog-react/src/hooks/useEntity.ts

```diff
@@ -49,6 +49,9 @@ export const useEntityFromUrl = (): EntityLoadingStatus => {
       errorApi.post(new Error('No name provided!'));
       navigate('/');
     }
+    if (!loading && !entity) {
```

Comment:
wonder if we want to check that there's no existing error here too?

Reply:
useAsync will return an error that is passed out, so it can be handled outside the hook. For the case of entity not being found, getEntityByName returns undefined instead of an error, so there is no error to handle.

Reply:
Right but `getEntityByName` CAN still fail for other reasons, and we don't want this error to be displayed when that happens. Because then we'll have loading=false, entity=undefined (which makes this if statement match), and error=something

So yes this should be `if (!loading && !error && !entity)`, because those are the precise circumstances when this error should be shown.

Reply:
However, I am not sure if 404s should be using the error API. The more common pattern, I think, is to show an actual 404 page. This is also what you do for techdocs below. So maybe this hook should not be changed, and instead look at [this file](https://github.com/backstage/backstage/blob/c654b10e91d1e75c663b2448e02af4d35f8cd48f/plugins/catalog/src/components/EntityLayout/EntityLayout.tsx#L223) and insert a

```tsx
      {!loading && !error && !entity && (
        <ErrorPage
          status="404"
          statusMessage="Not Found"
          additionalInfo="There is no catalog entity with that kind, namespace, and name"
        />
      )}
```

this is the `ErrorPage` exported from `core-components`.

Reply:
Updated